example/src/tests/groupPerformanceTests.ts

```diff
@@ -0,0 +1,442 @@
+/* eslint-disable @typescript-eslint/no-extra-non-null-assertion */
+import { Wallet } from 'ethers'
+import { Client, Group } from 'xmtp-react-native-sdk'
+
+import { Test, assert, createClients } from './test-utils'
+
+export const groupPerformanceTests: Test[] = []
+let counter = 1
+function test(name: string, perform: () => Promise<boolean>) {
+  groupPerformanceTests.push({
+    name: String(counter++) + '. ' + name,
+    run: perform,
+  })
+}
+
+async function createGroups(
+  client: Client,
+  peers: Client[],
+  numGroups: number,
+  numMessages: number
+): Promise<Group[]> {
+  const groups = []
+  const addresses: string[] = peers.map((client) => client.address)
+  for (let i = 0; i < numGroups; i++) {
+    const group = await client.conversations.newGroup(addresses, {
+      name: `group ${i}`,
+      imageUrlSquare: `www.group${i}.com`,
+      description: `group ${i}`,
+    })
+    groups.push(group)
+    for (let i = 0; i < numMessages; i++) {
+      await group.send({ text: `Alix message ${i}` })
+    }
+  }
+  return groups
+}
+
+async function createMessages(
+  group: Group,
+  numMessages: number,
+  name: string
+): Promise<number> {
+  let messages = 0
+  for (let i = 0; i < numMessages; i++) {
+    await group.send({ text: `${name} Message ${i}` })
+    messages++
+  }
+  return messages
+}
+
+test('testing small repro of odd streaming', async () => {
+  let groupCallbacks = 0
+  let messageCallbacks = 0
+  let boGroupCallbacks = 0
+  let boMessageCallbacks = 0
+
+  const [alixClient] = await createClients(1)
+  const peers = await createClients(20)
+  const boClient = peers[0]
+
+  await alixClient.conversations.streamGroups(async () => {
+    groupCallbacks++
+  })
+
+  await alixClient.conversations.streamAllMessages(async (message) => {
+    // console.log(`Alix recieved message ${message.content()}`)
+    messageCallbacks++
+  }, true)
+
+  await boClient.conversations.streamGroups(async () => {
+    boGroupCallbacks++
+  })
+
+  await boClient.conversations.streamAllMessages(async (message) => {
+    // console.log(`Bo recieved message ${message.content()}`)
+    boMessageCallbacks++
+  }, true)
+
+  const groups = await createGroups(alixClient, peers, 1, 10)
+
+  console.log(`Alix Streamed ${groupCallbacks} groups (1)`)
+  console.log(`Alix Streamed ${messageCallbacks} messages (10)`)
+
+  const alixGroup = groups[0]
+  const boGroup = await boClient.conversations.findGroup(alixGroup.id)
+
+  await createMessages(boGroup!!, 10, 'Bo')
+  await createMessages(alixGroup!!, 10, 'Alix')
+
+  console.log(`Alix Streamed ${groupCallbacks} groups (1)`)
+  console.log(`Alix Streamed ${messageCallbacks} messages (30)`)
+  console.log(`Bo Streamed ${boGroupCallbacks} groups (1)`)
+  console.log(`Bo Streamed ${boMessageCallbacks} messages (30)`)
+
+  return true
+})
```

Comment:
On this latest test I'm pretty consistently getting these errors

<details><summary>Details</summary>
<p>

```logs
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] error getting publish intent data UpdateGroupMembership(GroupStateError(PendingCommit))
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Sync: error publishing intents UpdateGroupMembership(GroupStateError(PendingCommit))
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] error syncing group Sync([UpdateGroupMembership(GroupStateError(PendingCommit))])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Wrong Epoch: message.epoch() 0 != 1 self.group_context().epoch()
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Error processing stream entry: Sync([UpdateGroupMembership(GroupStateError(PendingCommit))])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Wrong Epoch: message.epoch() 0 != 1 self.group_context().epoch()
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Message processing errors: [OpenMlsProcessMessage(ValidationError(WrongEpoch))]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] receive error ReceiveErrors([OpenMlsProcessMessage(ValidationError(WrongEpoch))])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 8
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516513] Message processing errors: [AlreadyProcessed(11656)]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516513] receive error ReceiveErrors([AlreadyProcessed(11656)])
 LOG  Alix Streamed 1 groups (10)
 LOG  Alix Streamed 10 messages (10)
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516472] Wrong Epoch: message.epoch() 0 != 1 self.group_context().epoch()
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516472] Message processing errors: [OpenMlsProcessMessage(ValidationError(WrongEpoch)),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id")))]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516472] receive error ReceiveErrors([OpenMlsProcessMessage(ValidationError(WrongEpoch)),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id")))])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516512] Message processing errors: [Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id")))]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516512] receive error ReceiveErrors([Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id")))])
 LOG  Alix Streamed 1 groups (10)
 LOG  Alix Streamed 30 messages (30)
 LOG  Bo Streamed 1 groups (10)
 LOG  Bo Streamed 30 messages (30)
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Error processing stream entry: Generic("Group intent could not be committed")
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Wrong Epoch: message.epoch() 0 != 1 self.group_context().epoch()
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Wrong Epoch: message.epoch() 0 != 1 self.group_context().epoch()
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Message processing errors: [OpenMlsProcessMessage(ValidationError(WrongEpoch))]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] receive error ReceiveErrors([OpenMlsProcessMessage(ValidationError(WrongEpoch))])
 LOG  Alix Streamed 1 groups (10)
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516477] Wrong Epoch: message.epoch() 0 != 1 self.group_context().epoch()
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516477] Message processing errors: [OpenMlsProcessMessage(ValidationError(WrongEpoch)),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id")))]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516477] receive error ReceiveErrors([OpenMlsProcessMessage(ValidationError(WrongEpoch)),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id")))])
 LOG  Alix Streamed 10 messages (10)
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14512341] Message processing errors: [Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id")))]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14512341] receive error ReceiveErrors([Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id")))])
 LOG  Alix Streamed 1 groups (1)
 LOG  Alix Streamed 30 messages (30)
 LOG  Bo Streamed 1 groups (1)
 LOG  Bo Streamed 30 messages (30)
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Error processing stream entry: Generic("Group intent could not be committed")
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Wrong Epoch: message.epoch() 0 != 1 self.group_context().epoch()
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Wrong Epoch: message.epoch() 0 != 1 self.group_context().epoch()
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Message processing errors: [OpenMlsProcessMessage(ValidationError(WrongEpoch))]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] receive error ReceiveErrors([OpenMlsProcessMessage(ValidationError(WrongEpoch))])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 4
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Message processing errors: [AlreadyProcessed(11714)]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] receive error ReceiveErrors([AlreadyProcessed(11714)])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 5
	RatchetTypeError
 LOG  Alix Streamed 1 groups (1)
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Message processing errors: [AlreadyProcessed(11719)]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] receive error ReceiveErrors([AlreadyProcessed(11719)])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 6
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 7
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 8
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 9
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14512341] Wrong Epoch: message.epoch() 0 != 1 self.group_context().epoch()
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14512341] Message processing errors: [OpenMlsProcessMessage(ValidationError(WrongEpoch)),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id")))]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14512341] receive error ReceiveErrors([OpenMlsProcessMessage(ValidationError(WrongEpoch)),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id")))])
 LOG  Alix Streamed 5 messages (10)
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 1
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14524663] Message processing errors: [AlreadyProcessed(11721)]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14524663] receive error ReceiveErrors([AlreadyProcessed(11721)])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 5
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14524663] Message processing errors: [Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id")))]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14524663] receive error ReceiveErrors([Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))),
Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE constraint failed: group_messages.id"))), Storage(DieselResult(DatabaseError(UniqueViolation, "UNIQUE
constraint failed: group_messages.id")))])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] Message processing errors: [AlreadyProcessed(11725)]
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] receive error ReceiveErrors([AlreadyProcessed(11725)])
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 6
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 7
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 8
	RatchetTypeError
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510] This is the wrong ratchet type.
[xmtpreactnativesdkexample] libxmtp[ERROR] - [libxmtp][t:14516510]   Ciphertext generation out of bounds 9
	RatchetTypeError
 LOG  Alix Streamed 1 groups (1)
 LOG  Alix Streamed 30 messages (30)
 LOG  Bo Streamed 1 groups (1)
 LOG  Bo Streamed 30 messages (30)
```


</p>
</details>

Reply:
Unique constraint reproduction in rust [here](https://github.com/xmtp/libxmtp/pull/958/files)

And I think the `Group intent could not be committed` is reproduced everytime with https://github.com/xmtp/libxmtp/issues/948